File: 53-maximum-subarray/Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        
        current = 0
        maxsum = nums[0]
        
        for i in range(len(nums)):
            current += nums[i]
            
            if current>maxsum:
                maxsum = current
            if current < 0:
                current = 0
            
        return maxsum
            

File: 53-maximum-subarray/Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums)
    {
        vector<int> memo(nums.size());

        memo[nums.size() - 1] = nums[nums.size() - 1];
        for (int i = nums.size() - 2;i >= 0; i--)
            memo[i] = max(nums[i], memo[i + 1] + nums[i]);
        return (*max_element(memo.begin(),memo.end()));
    }
};
